src/Infrastructure/IO/src/ESMCI_PIO_Handler.C

```diff
@@ -808,13 +812,24 @@ void PIO_Handler::arrayWrite(
   // Get a pointer to the array data
   // Still have the one DE restriction so use localDE = 0
   localDE = 0;
-  baseAddress = arr_p->getLocalarrayList()[localDE]->getBaseAddr();
-  PRINTMSG("baseAddress = 0x" << (void *)baseAddress);
-//  int arrlen = arr_p->getLocalarrayList()[localDE]->getByteCount();
-  int arrlen = 1;
-  const int *counts = arr_p->getLocalarrayList()[localDE]->getCounts();
-  for (int i=0; i<narrDims; i++)
-      arrlen *= counts[i]; 
+  // FIXME(wjs, 2022-07-22) Add loop over tiles
+  int tile = 1;
+  int tileOfThisDe = arr_p->getDistGrid()->getTilePLocalDe(localDE, &localrc);
+  if (ESMC_LogDefault.MsgFoundError(localrc, ESMCI_ERR_PASSTHRU,
+    ESMC_CONTEXT, rc)) return;
+  int arrlen;
+  if (tileOfThisDe == tile) {
+    baseAddress = arr_p->getLocalarrayList()[localDE]->getBaseAddr();
+    PRINTMSG("baseAddress = 0x" << (void *)baseAddress);
+    // arrlen = arr_p->getLocalarrayList()[localDE]->getByteCount();
+    arrlen = 1;
+    const int *counts = arr_p->getLocalarrayList()[localDE]->getCounts();
+    for (int i=0; i<narrDims; i++)
+      arrlen *= counts[i];
+  } else {
+    baseAddress = NULL;
```

Comment:
Passing a NULL to PIO from procs that don't own this tile feels like it will prevent inadvertently trying to read from / write to portions of memory that shouldn't be read from / written to for this call. However, I'm not 100% sure that this passing of NULL is the right thing to do in this situation. I'd like to confirm that, possibly with Jim Edwards.

Reply:
@jedwards4b can you please confirm Bill's logic on setting baseAddress to NULL when procs do not own a tile to be read/written?